plugin/someteam.example.com/v1/configmappatchjsontransformer/ConfigMapPatchJsonTransformer.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate pluginator
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/kustomize/api/ifc"
+	"sigs.k8s.io/kustomize/api/resid"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/yaml"
+)
+
+type stringMarshal struct {
+	Field string `yaml:",flow"`
+}
+
+type field struct {
+	decodedPatch jsonpatch.Patch
+	Name         string `json:"name" yaml:"name"`
+	Path         string `json:"path,omitempty" yaml:"path,omitempty"`
+	JsonOp       string `json:"jsonOp,omitempty" yaml:"jsonOp,omitempty"`
+}
+
+type plugin struct {
+	ldr    ifc.Loader
+	Target types.PatchTarget `json:"target,omitempty" yaml:"target,omitempty"`
+	Fields []*field          `json:"fields,omitempty" yaml:"fields,omitempty"`
+}
+
+//noinspection GoUnusedGlobalVariable
+var KustomizePlugin plugin
+
+func (p *plugin) Config(
+	h *resmap.PluginHelpers, c []byte) (err error) {
+	p.ldr = h.Loader()
+	err = yaml.Unmarshal(c, p)
+	if err != nil {
+		return err
+	}
+	if p.Target.Name == "" {
+		return fmt.Errorf("must specify the target name")
+	}
+
+	for _, field := range p.Fields {
+		if lerr := p.load(field); lerr != nil {
+			return lerr
+		}
+	}
+
+	return err
+}
+
+func (p *plugin) Transform(m resmap.ResMap) error {
+	id := resid.NewResIdWithNamespace(
+		resid.Gvk{
+			Group:   p.Target.Group,
+			Version: p.Target.Version,
+			Kind:    p.Target.Kind,
+		},
+		p.Target.Name,
+		p.Target.Namespace,
+	)
+	obj, err := m.GetById(id)
+	if err != nil {
+		return err
+	}
+	if obj.GetKind() != "ConfigMap" {
+		return fmt.Errorf("This patch cannot be applied to non configmap resources")
+	}
+
+	configMapData, gerr := obj.GetFieldValue("data")
```

Comment:
Can you use `kyaml/yaml` library to manipulate resources? You can view the example in `api/builtins/PatchJson6902Transformer.go`

Reply:
@Shell32-Natsu I'm using the exact same imports as the builtin patchJson6902Transformer. That isnt sufficient ?

Reply:
@mortent

Reply:
@Shell32-Natsu Just updated to use kyaml